src/lib/encoder.ts

```diff
@@ -129,12 +133,33 @@ const decodeDataSourceWithHash = (value: string): URLDataWithHash => {
   const verificationMethodSignature = value.slice(0, 10);
   const verificationMethod = getVerificationMethod(verificationMethodSignature);
   const encodedData = value.slice(10); // Rest of data string after function hash
+
+  try {
+    const dataSource = hexToUtf8('0x' + encodedData); // Get as URI
+    if (encodedData.length < 64 || /^(https?|ipfs):\/\//.test(dataSource)) {
+      console.log(dataSource);
+      return {
+        verification: {
+          method: NONE_VERIFICATION_METHOD,
+          data: '0x',
+        },
+        url: dataSource,
+      };
+    }
+  } catch {
+    // ignore
+  }
+
   const dataHash = '0x' + encodedData.slice(0, 64); // Get jsonHash 32 bytes
   const dataSource = hexToUtf8('0x' + encodedData.slice(64)); // Get remainder as URI
 
   return {
     verification: {
-      method: verificationMethod?.name || UNKNOWN_VERIFICATION_METHOD,
+      method:
+        verificationMethod?.name ||
+        (verificationMethodSignature === '0x00000000'
```

Comment:
```suggestion
        (verificationMethodSignature === NONE_VERIFICATION_METHOD
```

Reply:
veirifacrionMethodSignature === "0x0000..." will return a bool right? Do we want a book here in method?

Reply:
It is now, you're right. Initially the NONE_VERIFICATION_METHOD was an empty string, but it turns out that caused problems elsewhere. Fixing.

Reply:
Simplified to just return the verificationMethodSignature.
The "unknown" thing was what was causing problems because as a user of erc725 it was not possible to determine what the signature was for a custom verification method and 0x0000000 would also be marked as unknown.